### ISSUE TITLE
Add npm for project build and pnpm for monorepo management

### ISSUE THREAD

**Issue by joaquimrocha** [kubernetes-sigs/headlamp#2206]
We have currently make and npm, where make just calls npm and go for building the frontend and backend, respectively.
While this is common for many Linux developers, it's not common for many Js ones, and there's no real reason why make is needed. So we should add a packages.json to the root folder of the project and use it to build the project just having node + go.

Since we have also 3 npm projects in Headlamp (frontend, app, and headlamp-plugin, not counting the example plugins), we could also a tool for managing monorepos, like pnpm. Same for the plugins repo.

- [ ] Make the project buildable by just using npm run scripts
- [ ] Set the Makefile as deprecated (print warnings in commands)
- [ ] Use pnpm in order to manage the different build options and npm sub-projects (headlamp-plugin, frontend, app)

**Comment by sniok** [comment]
I agree, monorepo setup makes a lot of sense for us. Although I don't think lerna is a good choice. It used to be a go-to for monorepos but package managers have this functionality built-in now, see
[npm workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces) or [pnpm workspace](https://pnpm.io/workspaces). The latter is actually gaining a lot of popularity because of how much more faster and efficient it is over npm. Projects like material ui, vite, next.js use it for their monorepos. Lerna still has some more features on top but there are alternative tools that might be better, like [turborepo](https://turbo.build/repo/docs) (not to be confused with turbopack) and [nx](https://nx.dev/).

I think we should consider pnpm + turborepo
